refine(undelegation): add a parameter to limit the number of undelegation completions

TimmyExogenous opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added max_undelegation_completions parameter to limit undelegation completions processed per block (0 = no limit)

* **CLI**
  * CLI command renamed/generalized to update params and now accepts max_undelegation_completions

* **Tests**
  * Added tests covering no limit, partial limit, and full-limit undelegation completion scenarios

* **Chores**
  * API docs and responses updated to expose the new parameter
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

TimmyExogenous committed:
add a parameter to limit the number of undelegation completions per block

TimmyExogenous commented on client/docs/swagger-ui/swagger.json:
```diff
@@ -23658,6 +23658,11 @@
                       "format": "int64",
                       "title": "instant_undelegation_penalty is the penalty percentage for instant undelegation (e.g. 25 for 25%)"
+                    },
+                    "max_undelegation_completions": {
+                      "type": "integer",
+                      "format": "int64",
+                      "description": "max_undelegation_completions is the maximum number of undelegation completions processed per block.\nA value of 0 indicates no limit."
```
I've already updated the Swagger file. It looks like an issue with the generation tool.

MaxMustermann2 reviewed (commented):
Please add a test and include in MsgUpdateParams

TimmyExogenous committed:
add unit test for MaxUndelegationCompletions param

TimmyExogenous commented:
> Please add a test and include in MsgUpdateParams

The test has been added.

bwhour reviewed (approved):
the whole looks good. but better documented CLI changes.

TimmyExogenous committed:
Merge remote-tracking branch 'upstream/develop' into limit-undelegation-completion

TimmyExogenous committed:
Merge remote-tracking branch 'upstream/develop' into limit-undelegation-completion

cloud8little committed:
Merge branch 'develop' into limit-undelegation-completion

TimmyExogenous committed:
fix(delegation): address PR #399 review comments; add cursor rules

Code changes:
- Fix pointer alias: refine proto (gogoproto.nullable=false), update keeper and tests
- Remove redundant early return in GetCompletableUndelegations (loop break suffices)
- Add CLI Long description for UpdateParams (instant-undelegation-penalty, max-undelegation-completions)
- Makefile: add --network host for proto Docker (buf BSR access)

Cursor rules:
- git-conventions: default branch, naming, PR targeting
- proto-generated-files: do not edit *.pb.go, proto-gen workflow

Made-with: Cursor

TimmyExogenous committed:
chore(rules): refine pre-push lint/test to use judgment based on change scale

Made-with: Cursor

TimmyExogenous committed:
Merge branch 'limit-undelegation-completion' of TimmyExogenous-github.com:TimmyExogenous/imuachain into limit-undelegation-completion

TimmyExogenous committed:
fix(delegation): use value type for Undelegation at missed call sites

Address remaining usages after proto change (gogoproto.nullable=false):
- delegation.go: Undelegation: &r -> Undelegation: r
- assets_test.go: Undelegation: &... -> Undelegation: ...

Made-with: Cursor
